test/EVMHost.cpp

```diff
@@ -167,11 +169,21 @@ void EVMHost::selfdestruct(const evmc::address& _addr, const evmc::address& _ben
 	accounts.erase(_addr);
 	accounts[_beneficiary].balance = balance;
 	// Record self destructs
-	recorded_selfdestructs.push_back({_addr, _beneficiary});
+	recorded_selfdestructs.push_back({_addr, _beneficiary, balance});
+}
+
+void EVMHost::recordCalls(evmc_message const& _message) noexcept
+{
+	if (recorded_calls.empty())
+		recorded_calls.reserve(max_recorded_calls);
```

Comment:
Should this rather be ``if(!recorded_calls.capacity())``?
Also do we want to run it here every time anyways?
Seems to me that, if at all, this should be done in the constructor and probably the constructor of `MockedHost` at that... note that `.clear()` on a vector does not reset its capacity.

Reply:
Thank you. Fixed this.